.github/workflows/createNewVersion.yml

```diff
@@ -0,0 +1,124 @@
+    runs-on: macos-latest
+
+    outputs:
+      PULL_REQUEST_NUMBER: ${{ steps.createPullRequest.outputs.pull-request-number }}
+
+    steps:
+      # Version: 2.3.4
+      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.OS_BOTIFY_TOKEN }}
+
+      - uses: softprops/turnstyle@8db075d65b19bf94e6e8687b504db69938dc3c65
+        with:
+          poll-interval-seconds: 10
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set up git
+        run: |
+          git fetch
+          git checkout main
+          git config user.name OSBotify
+
+      - name: Create new branch
+        run: |
+          git checkout -b version-${{ github.event.inputs.SEMVER_LEVEL }}-${{ github.sha }}
+          git push --set-upstream origin version-${{ github.event.inputs.SEMVER_LEVEL }}-${{ github.sha }}
+
+      - name: Generate version
+        id: bumpVersion
+        uses: Expensify/Expensify.cash/.github/actions/bumpVersion@main
+        with:
+          GITHUB_TOKEN: ${{ secrets.OS_BOTIFY_TOKEN }}
+          SEMVER_LEVLE: ${{ github.event.inputs.SEMVER_LEVEL }}
+
+      - name: Commit new version
+        run: |
+          git add \
+            ./package.json \
+            ./package-lock.json \
+            ./android/app/build.gradle \
+            ./ios/ExpensifyCash/Info.plist \
+            ./ios/ExpensifyCashTests/Info.plist
+          git commit -m "Update version to ${{ steps.bumpVersion.outputs.NEW_VERSION }}"
+
+      - name: Create Pull Request (main)
+        id: createPullRequest
+        uses: peter-evans/create-pull-request@09b9ac155b0d5ad7d8d157ed32158c1b73689109
+        with:
+          token: ${{ secrets.OS_BOTIFY_TOKEN }}
+          author: OSBotify <reactnative@expensify.com>
+          base: main
+          branch: version-${{ github.event.inputs.SEMVER_LEVEL }}-${{ github.sha }}
+          title: Update version to ${{ steps.bumpVersion.outputs.NEW_VERSION }} on main
+          body: Update version to ${{ steps.bumpVersion.outputs.NEW_VERSION }}
+
+      # This Slack step is duplicated in all workflows, if you make a change to this step, make sure to update all
+      # the other workflows with the same change
+      - uses: 8398a7/action-slack@v3
+        name: Job failed Slack notification
+        if: ${{ failure() }}
+        with:
+          status: custom
+          fields: workflow, repo
+          custom_payload: |
+            {
+              channel: '#announce',
+              attachments: [{
+                color: "#DB4545",
+                pretext: `<!here>`,
+                text: `💥 ${process.env.AS_REPO} failed on ${process.env.AS_WORKFLOW} workflow 💥`,
+              }]
+            }
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK }}
+
+  # Note: This is a separate job because trigger-workflow-and-wait doesn't work on a macOS runner
```

Comment:
Bummer - Are there any alternatives that do? How hard would it be to write our own in JS?

Reply:
Funny enough, there is a quirk with this in that the `GITHUB_TOKEN` cannot be used to trigger new actions (by design, to prevent infinite loops creating workflows). Before I realized that the workaround is to simply use a PAT (`OS_BOTIFY_TOKEN` in our case), I sort of wrote it off as a problem with the `trigger-workflow-and-wait` action. So I started writing my own, and have a [draft PR](https://github.com/Expensify/Expensify.cash/pull/2495). You can take a look ... it's a bit complex.

Reply:
So.. Should we use yours since it can work on mac-os and will simplify this workflow code at least?

Reply:
+1, I think we should look into using the draft PR implementation for this then

Reply:
Okay, in that case I will need to polish and test that draft PR. I believe it can be tested in `Public-Test-Repo` to validate that it works.